### ISSUE TITLE
Unclosed quote error observed when all quotes are properly balanced

### ISSUE THREAD

**Issue by cookch10** [postcss/postcss-scss#30]
Originally opened in https://github.com/stylelint/stylelint/issues/1150.

When using stylelint `v6.2.2` as follows: 

package.json (simplified):
```json
{
  "scripts": {
    "lint:styles": "stylelint --syntax scss \"src/**/*.scss\""
  }
}
```

My .stylelintrc config (uses stylelint-config-standard `v6.0.0`):
```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "function-url-quotes": "none",
    "selector-pseudo-element-colon-notation": "single"
  }
}
```

given the following Sass (scss file):
```scss
@import "../../../styles/config";

.some-widget {
    $column: #{"#{&}__column"};

    > .row {
        > #{$column}__section {
            .some__section {
                @include some-breakpoint-mixin("desktop") {
                    display: none;
                }
            }
        }
    }
}
```

```shell
npm run lint:styles
```

The following results in an error:
```shell
 9:49  ×  Unclosed quote   CssSyntaxError
```

This is valid Sass and should not be yielding an unclosed quote error.

**Follow-up:**  If I change the following:

```scss
$column: #{"#{&}__column"};
```

to

```scss
$column: "column";
```

The Unclosed quote error is not thrown.  So it seems there may be an issue with parsing quotes in interpolated Sass strings / variables.

**Comment by ai** [comment]
I know a source: seems like wrong logic with recursive `#{}`

Does recursive `#{}` is a valid thing in SCSS?

**Comment by evilebottnawi** [comment]
@ai yep, 
```scss
.some-widget {
    $column: #{"#{&}__column"};

    > .row {
        > #{$column}__section {
            .some__section {
                color: red;
            }
        }
    }
}
```
to
```css
.some-widget > .row > .some-widget__column__section .some__section {
  color: red;
}
```

**Comment by ai** [comment]
OK, I will try to look on this week.

**Comment by cookch10** [comment]
Thank you @ai.

**Comment by cookch10** [comment]
@ai, Have you had a chance to look into this?

**Comment by ai** [comment]
Nope 😢 Sorry, I am preparing for talk at May 11.

Feel free to ping me in May 13.

Also I can always accept PR (but I understand that fixing parser is hard thing).

**Comment by cookch10** [comment]
No worries :)

Was considering a PR, but to your point, since it is with parser, I figured you probably have the best context on what / where updates need to be applied.  I'll check back later.  Good luck with your talk!

**Comment by ai** [comment]
Done 41cf69a

**Comment by evilebottnawi** [comment]
@ai Thanks

**Comment by ai** [comment]
Released in 0.1.8.

**Comment by cookch10** [comment]
Awesome.  Thanks @ai !
